pkg/app/AppService.go

```diff
@@ -166,6 +167,138 @@ func (impl AppServiceImpl) UpdateReleaseStatus(updateStatusRequest *bean.Release
 	return count == 1, nil
 }
 
+
+func (impl AppServiceImpl) GetDeploymentStatus(appName string, appId, envId int) (*bean.DeploymentStatus, error) {
+	pipelineOverride, err := impl.pipelineOverrideRepository.GetLatestRelease(appId, envId)
+	if err != nil {
+		return nil, err
+	}
+	deploymentStatus := &bean.DeploymentStatus{LastDeploymentStatus: bean.StepStatusProgress}
+	//init with default status
+	waitingStatus := &bean.StepDetail{Status: bean.StepStatusWaiting}
+	deploymentStatus.GitPushStep = waitingStatus
+	deploymentStatus.GitPullStep = waitingStatus
+	deploymentStatus.K8sDeploy = waitingStatus
+	deploymentStatus.ConfigApplyStep = waitingStatus
+
+	//git push
+	deploymentStatus.ReleaseHash = pipelineOverride.GitHash
+	deploymentStatus.ReleaseTime = pipelineOverride.CreatedOn
+	gitPush := &bean.StepDetail{}
+	gitPushError := pipelineOverride.Status == models.CHARTSTATUS_ERROR
+	if pipelineOverride.GitHash != "" {
+		gitPush.Status = bean.StepStatusSuccess
+	} else {
+		if gitPushError {
+			gitPush.Status = bean.StepStatusError
+			gitPush.ErrorMessage = pipelineOverride.ErrorMsg
+		} else {
+			gitPush.Status = bean.StepStatusProgress
+		}
+	}
+	deploymentStatus.GitPushStep = gitPush
+	if gitPush.Status != bean.StepStatusSuccess {
+		return deploymentStatus, nil
+	}
+	//fetch acd app
+	appString, err := impl.ArgoK8sClient.GetApplication("devtroncd", appName)
+	if err != nil {
+		impl.logger.Errorw("error is getting acd app", "name", appName, "err", err)
+		return nil, err
+	}
+	var application v1alpha1.Application
```

Comment:
argo-cd 2.0 has v1

Reply:
we should use unstructured.unstructured so that we dont depend on application library